Core/Core/Notifications/NotificationManager.swift

```diff
@@ -85,6 +85,9 @@ extension NotificationManager {
     }
 
     public func subscribeToPushChannel(token: Data? = nil, session: LoginSession? = AppEnvironment.shared.currentSession) {
+        guard AppEnvironment.shared.currentSession?.masquerader?.absoluteString.contains("fake-students") == nil else {
```

Comment:
There is an `isFakeStudent` method on `LoginSession`, can't we use that?

Reply:
done